### DISCUSSION TITLE
CTF4: CodeQL and chill: community feedback

### DISCUSSION THREAD

**Discussion by xcorail** [github/securitylab#141] [General] [answered]
👋🏾 dear CTF participants!

I hope you enjoyed this CTF! 
Several of you have asked for feedback on their submissions, so I thought it would be a good idea to publish your solutions here, and get feedback from the community! 

If you want this feedback, please make your solution public and post it here (start a new thread for each, so that we know which feedback is for which).

For info, the model answers will be published on Tuesday.

Cheers!

**Comment by atorralba** [comment]
Amazing initiative, thanks! Here's my write-up, don't hesitate to criticize it to your heart's content :-)

https://github.com/atorralba/GHSL_CTF_4

**Comment by kanav99** [comment]
Here's mine! https://github.com/kanav99/github-java-ctf , would really like some feedback on the bonus section!

**Comment by cldrn** [comment]
Thanks a lot for the additional feedback! I was looking for the model answers to see where I can improve but this is even better: [https://github.com/cldrn/ctf4-codeql-and-chill-java](https://github.com/cldrn/ctf4-codeql-and-chill-java)

**Comment by sin3point14** [comment]
I don't have anything unique to say 😅 so just gonna drop this here
https://github.com/sin3point14/githubctf2020

**Comment by adityasharad** [comment, marked as answer]
Thanks for participating and for sharing your solutions! Many of you asked for the reference answers -- these are now published at https://securitylab.github.com/ctf/codeql-and-chill/answers. One of the great things about CodeQL is that there are many elegant ways to solve the same problem, so please remember that this is just one possible approach and we hope that you find it a useful reference.

**Comment by h3ku** [comment]
Here is mine, I'm enjoying reading all the other ones especially the details I missed about the exploitation.

https://github.com/h3ku/CodeQLCTF4

**Comment by paraschetal** [comment]
Thanks for organizing the CTF and the reference solution. Here's my write-up: https://github.com/paraschetal/codeql-and-chill
I'll appreciate any feedback :)

**Comment by osynetskyi** [comment]
Adding my writeup to the pile! :)
[https://gist.github.com/osynetskyi/72cb82011e28549c4a5bedc2437dffa9](https://gist.github.com/osynetskyi/72cb82011e28549c4a5bedc2437dffa9)
